### PR TITLE
AM-4237 SSL SNI support

### PR DESCRIPTION
**Pull request by enriquedacostacambio**
https://appdirect.jira.com/browse/AM-4237

Adds SNI support to openid4java and refactors the way SNI support was added to CXF.

cc @yanavery @malacroix

### CODE REVIEW

File: sample-isv-web/src/main/java/com/appdirect/isv/integration/util/ssl/SniSslSocket.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+import lombok.RequiredArgsConstructor;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.collect.ImmutableList;
+
+@RequiredArgsConstructor
+public class SniSslSocket extends SSLSocket {
+	private final SSLSocket delegate;
```
Comment (yanavery) [inline sample-isv-web/src/main/java/com/appdirect/isv/integration/util/ssl/SniSslSocket.java]: Where is ```delegate``` being set? Lomboked constructor or should that be done explicitly as part of the ```prepare()``` method?

File: sample-isv-web/src/main/java/com/appdirect/isv/integration/util/ssl/SniSslSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import java.net.InetAddress;
+import java.net.Socket;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class SniSslSocketFactory extends SSLSocketFactory {
+	private final SSLSocketFactory delegate;
```
Comment (yanavery) [inline sample-isv-web/src/main/java/com/appdirect/isv/integration/util/ssl/SniSslSocketFactory.java]: This ```delegate``` I see is set via constructor ... 👍

**Comment by yanavery** [comment]
LGTM.

**Comment by yanavery** [comment]
@enriquedacostacambio, please add TestRail test case and run automation regression against PR. 

Thanks!

**Comment by enriquedacostacambio** [comment]
@yanavery given this is a sample-isv PR i don't think there is a regression to run. do you think we need TCs for this project?

**Comment by yanavery** [comment]
@enriquedacostacambio, you're right, I had not even noticed. All good. Thanks!

**Pull request merged**